central/imageintegration/datastore/singleton.go

```diff
@@ -75,6 +78,65 @@ func deleteAutogeneratedRegistries(ctx context.Context, iiStore store.Store, iis
 	log.Info("Completed deletion of image integrations")
 }
 
+// setupScannerV4Integration will ensure the default Scanner V4 integration exists if
+// Scanner V4 is installed/enabled (forces opt-in).  If Scanner V4 is not installed (or is disabled)
+// will delete all ScannerV4 integrations.
+func setupScannerV4Integration(ctx context.Context, iiStore store.Store, iis []*storage.ImageIntegration) {
+	keepDefault := true // for readability.
+
+	// If Scanner V4 is not installed delete the associated integration(s).
+	if !features.ScannerV4Enabled.Enabled() {
+		deleteScannerV4Integrations(ctx, iiStore, iis, !keepDefault)
+		return
+	}
+
+	// Create the default Scanner V4 integration if it doesn't exist.
+	createScannerV4Integration(ctx, iiStore)
+
+	// Delete all but the default Scanner V4 integration (should be none).
+	deleteScannerV4Integrations(ctx, iiStore, iis, keepDefault)
+}
+
+// createScannerV4Integration will create the default Scanner V4 integration if it does
+// not currently exist.
+func createScannerV4Integration(ctx context.Context, iiStore store.Store) {
+	if _, exists, err := iiStore.Get(ctx, store.DefaultScannerV4Integration.GetId()); err != nil {
+		log.Errorf("Unable to detect if default Scanner V4 integration exists: %v", err)
+		return
```

Comment:
Do you think we should just add it, if we don't know if it exists or not? Is there any harm in upserting it again since it's immutable (at this time)

Reply:
It is mutable, for example the `num_concurrent_scans` can be changed at runtime. 

The indexer/matcher URLs can be changed as well. I'd be OK if we want to add logic to prevent the URLs from being changed, my thoughts for keeping those mutable was so we have a way out if the default's are ever wrong.